contributors/devel/api_changes.md

```diff
@@ -442,43 +447,39 @@ function converting type X in pkg a to type Y in pkg b, should be named:
 Also note that you can (and for efficiency reasons should) use auto-generated
 conversion functions when writing your conversion functions.
 
+Unsurprisingly, adding manually written conversion also requires you to add
+tests to `pkg/apis/<group>/<version>/conversion_test.go`.
+
 Once all the necessary manually written conversions are added, you need to
 regenerate auto-generated ones. To regenerate them run:
 
 ```sh
-hack/update-codegen.sh
+make clean && make generated_files
 ```
 
-As part of the build, kubernetes will also generate code to handle deep copy of
-your versioned api objects. The deep copy code resides with each versioned API:
-   - `<path_to_versioned_api>/zz_generated.deepcopy.go` containing auto-generated copy functions
+`make clean` is important, otherwise the generated files might be stale, because
+the build system uses custom cache.
+
+`make all` will invoke `make generated_files` as well.
+
+The `make generated_files` will also regenerate the `zz_generated.deepcopy.go`,
+`zz_generated.defaults.go`, and `api/openapi-spec/openapi-spec`.
 
 If regeneration is somehow not possible due to compile errors, the easiest
 workaround is to comment out the code causing errors and let the script to
 regenerate it. If the auto-generated conversion methods are not used by the
 manually-written ones, it's fine to just remove the whole file and let the
 generator to create it from scratch.
 
-Unsurprisingly, adding manually written conversion also requires you to add
-tests to `pkg/api/<version>/conversion_test.go`.
-
 ## Generate Code 
 
-When any `types.go` change is made, the generated code must be updated
-by running the generators. There are many small generators that
-run. They have some dependencies on each other so they must be run in
```

Comment:
I removed the words on "dependencies" because it's handled by hack/update-codegen.sh"